### ISSUE TITLE
Could not convert parameter `tx` between node and runtime: No such variant in enum MultiAddress

### ISSUE THREAD

**Issue by boohyung** [polkadot-js/api#3216]
The sending error is remaining after applying #3215 (v3.10.3-6).

**Pieces of code**
> ApiPromise.create({ provider, types: {      Balance: 'u64'    } }).then(async a => {
    PolkadotSigner.api.registerTypes({ Address: "AccountId", LookupSource: "AccountId" })
}

**error message**
> 1002: Verification Error: Runtime error: Execution failed: ApiError("Could not convert parameter `tx` between node and runtime: No such variant in enum MultiAddress"): RuntimeApi, Execution failed: ApiError, \"Could not convert parameter `tx` between node and runtime: No such variant in enum MultiAddress\

I think it's still a multiaddress type. What should I change to solve the problem?

**Comment by jacogr** [comment]
Just remove the `{ Address: "AccountId", LookupSource: "AccountId" }` types. Basically since the chain complains about `MultiAddress` it does mean it uses it. (So removing the above should just apply the defaults)

**Comment by boohyung** [comment]
I tried to remove the { Address: "AccountId", LookupSource: "AccountId" } as you comment. But another error is found.

{"error":"createType(Address):: Unable to create Enum via index 206, in Id, Index, Raw, Address32, Address20"}

I want to know about meaning of index and solve the problem.

**Comment by jacogr** [comment]
It will only work in the latest beta e.g. 2.10.3-6 or later, e.g. after https://github.com/polkadot-js/api/pull/3215

**Comment by boohyung** [comment]
There are simple sending transaction logic and test code. but the problem is not solved.

**logic**
ApiPromise.create -> new Keyring -> createRawTransaction -> signRawTransaction -> sendRawTransaction

**error**
{"error":"createType(Address):: Unable to create Enum via index 206, in Id, Index, Raw, Address32, Address20"}

**Test Code**
```javascript
import { Keyring, ApiPromise, WsProvider } from '@polkadot/api';

const substrateNetworkBase = {
[substrateNetworkKeys.POLKADOT]: {
    decimals: 10,
    genesisHash: null,
    order: 1,
    pathId: "polkadot",
    prefix: 0,
    title: "Polkadot",
    unit: "DOT",
  },
[substrateNetworkKeys.WESTEND]: {
    decimals: 12,
    genesisHash: substrateNetworkKeys.WESTEND,
    order: 3,
    pathId: "westend",
    prefix: 42,
    title: "Westend",
    unit: "WND",
  },
};

const provider = new WsProvider(`${config.provider.hostUrl}${config.provider.port ? `:${config.provider.port}` : ''}`);

ApiPromise.create({ provider }).then(async a => {
      PolkadotSigner.api = a
      this.genesisHash = await (await a.rpc.chain.getBlockHash(0)).toString()
      this.keyring = new Keyring({ type: 'sr25519', ss58Format: substrateNetworkBase[this.genesisHash].prefix }); //ss58Format setting
    })

...

createRawTransaction(txParams: ITransaction): any {
    txParams.transfer = PolkadotSigner.api.tx.balances.transfer(txParams.to, this.convertUnit("to", txParams.amount)) 
    return txParams
  }

  async signRawTransaction(txParams: any): Promise<any> {
    const nonce = await this.getNonce(txParams.from)
    return await txParams.transfer.sign(this.wallet[txParams.from], { nonce })  // signedTx
  }

  async sendRawTransaction(signedTx): Promise<string> {
    
    const signer = signedTx.signer.toHuman()
    const nonce = await this.getNonce(signer)
    await signedTx.send(this.wallet[signer], { nonce }) 
    const buffer = Buffer.from(signedTx.hash)
    return `0x${buffer.toString('hex')}` 
  }
```

**Comment by jacogr** [comment]
I've used your approach with sign/send (and commented on areas as to why not to). Executed via -

- `mkdir -p test`
- `cd test`
- `yarn init`
- `yarn add @polkadot/api@beta`
- `node ./index.mjs`

```js
import { ApiPromise, Keyring } from '@polkadot/api';

async function main () {
  const api = await ApiPromise.create();
  const keyring = new Keyring({ type: 'sr25519' });
  const alice = keyring.addFromUri('//Alice');
  const bob = keyring.addFromUri('//Bob');

  // create the to tx
  const tx = api.tx.balances.transfer(bob.address, 123456789876n);

  // get the nonce (since we are not using the pass-through signAsync - see comments below)
  const { nonce } = api.query.system.account(alice.address);

  // sign this is sync (& deprecated, use signAsync - which also gets all the mortality info and no
  // need to pass the nonce as a param, it will also take the pool into account)
  // NOTE: In this case we will create an immortal since non of the era information is available
  tx.sign(alice, { nonce });

  // send, has no params (only awaiting the hash)
  const hash = await tx.send();

  console.log(hash.toHex());
}

main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(-1);
  });
```

So not sure where your conversions go wrong, but indeed it works as expected.

**Comment by boohyung** [comment]
Thanks again @jacogr but same error when I execute `sign()`
I found same error when I did `signAsync `instead of `sign()`.

```javascript
createRawTransaction(txParams: ITransaction): any {
  txParams.transfer = PolkadotSigner.api.tx.balances.transfer(txParams.to, this.convertUnit("to", txParams.amount))
  
  return txParams
}

async signRawTransaction(txParams: any): Promise<any> {

    this.logger.debug('raw tx:', txParams.transfer.toString()); // raw tx: {"signature":{"signer":{"Id":"5C4hrfjw9DjXZTzV3MwzrrAr9P1MJhSrvWGWqi1eSuyUpnhM"},"signature":{"Ed25519":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"},"era":{"ImmortalEra":"0x00"},"nonce":0,"tip":0},"method":{"callIndex":"0x0400","args":{"dest":{"Id":"5F3nmjeRb3PDRQ7EagnsDPFhnsbEQwmqLjz5cT1X9Rb9RR8y"},"value":10000000000}}}

    const { nonce } = await PolkadotSigner.api.query.system.account(this.wallet[txParams.from].address)
    this.logger.debug('nonce:', nonce); // BN: 3
    
    const signedTx = await txParams.transfer.sign(this.wallet[txParams.from], { nonce })
    this.logger.debug(signedTx.toString()) // REGISTRY: Error: Unable to create Enum via index 206, in Id, Index, Raw, Address32, Address20
}


```

**Comment by jacogr** [comment]
That only means your dependencies are not where you think they are on the beta.

So rather then (without knowing what is in your package.json), explicitly ensure that your api & types match (so `yarn add @polkadot/api@beta @polkadot/types@beta` - if types has not bee updates alongside API, and metadata, etc. it won't work)

**Comment by boohyung** [comment]
I have updated `@polkadot/api@beta & @polkadot/types@beta` as you comment.

so there are added to my `package.json` 
- "@polkadot/api": "^3.10.3-10"
-  "@polkadot/types": "^3.10.3-10"

but another error is occured -> API/INIT: Error: You need to pass a valid Metadata instance to Decorated

**Comment by jacogr** [comment]
`@polkadot/metadata@beta`

**Comment by boohyung** [comment]
The same error is found after `yarn add @polkadot/metadata@beta`

**Comment by jacogr** [comment]
This "You need to pass a valid Metadata instance to Decorated"  is caused by the metadata version in the API & types not aligning.

When updating the API -

- everything `@polakdot/api-*` needs to be the same (this is contracts, derives, api, etc.)
- everything `@polkadot/types-` needs to be the same, this is known types, types
- metadata needs to align with the above

TL;DR You cannot mix versions between.

**Comment by boohyung** [comment]
Thanks @jacogr. "You need to pass a valid Metadata instance to Decorated" error no longer occurs. 
But I have one more question for you. How do you call the `send()` function from the `sendRawTransaction` function that receives the `signedTx `object as an argument?

My code is as below.
```javascript
async sendRawTransaction(signedTx): Promise<string> {
  
  const signer = signedTx.signer.toHuman() // { Id: '5GjSoc5PYRoJef7xSudicYR8JmncPSkY9zNN56ZFizM2cYSa' }
  const nonce = await this.getNonce(signer) // 3
  await signedTx.send(this.wallet[signer], { nonce }) 
  const buffer = Buffer.from(signedTx.hash)
  return `0x${buffer.toString('hex')}` 
}
```

**Comment by jacogr** [comment]
`.send()` does not take the signer (it is already signer), not options such as nonce (these are applied at the time of signing). See https://github.com/polkadot-js/api/blob/master/packages/api/src/submittable/types.ts#L56-L58

**Comment by boohyung** [comment]
Thanks for your support @jacogr . I can solve this problem :D
